tests/test-dladdr.c

```diff
@@ -0,0 +1,171 @@
+        if( verbose || !passed )
+            fprintf( stderr,"(could not get symbol information for address %p)\n", addr );
+        return !passed;
+    }
+    else
+    {
+        int sym_match  = info.dli_sname && strcmp( addrsym, info.dli_sname ) == 0;
+        int addr_match  = addr == info.dli_saddr;
+        passed = (expected_result == Pass && sym_match && addr_match)
+                 || (expected_result == PassWithoutSymbol && addr_match)
+                 || (expected_result == PassWithDifferentAddress && sym_match)
+                 || (expected_result == Fail && !(sym_match && addr_match));
+        printf( "check address %p with has symbol '%s' -> %s%s",addr, addrsym, passed ? "passed" : "failed", verbose || !passed ? " " : "\n" );
+        if( verbose || !passed )
+            print_dl_info( &info, "(", ")\n");
+        return !passed;
+    }
+}
+
+/**
+ * @brief return address from a symbol located in a shared lilbrary
+ * @param libname librray to get the address from
+ * @param addrsym symbol to get the address for
+ * @return pointer to symbol address or NULL
+ */
+void *get_symbol_address( char *libname, char *sym )
+{
+    void *library = NULL;
+    void *addr = NULL;
+
+    library = dlopen( libname, RTLD_GLOBAL );
+    if ( library == NULL )
+    {
+        fprintf( stderr, "could not open '%s'\n", libname );
+        return NULL;
+    }
+
+    addr = dlsym( library, sym );
+    dlclose( library );
+    if (!addr)
+        fprintf( stderr, "could not get address for symbol '%s'\n", sym );
+    return addr;
+}
+
+/**
+ * @brief check address from a symbol located in a shared lilbrary
+ * @param libname librray to get the address from
+ * @param addrsym symbol to get the address for
+ * @param should_match result should match the given values
+ * @return 0 check passed
+ * @return 1 check failed
+ */
+int check_dladdr_by_dlopen( char *libname, char *sym, int should_match )
+{
+    void *addr = get_symbol_address( libname, sym);
+    if (!addr)
+        return 1;
+    return check_dladdr( addr, sym, should_match );
+}
+
+#ifdef _WIN32
+#define HMODULE void*
+#define HANDLE void*
+#define LPCVOID void*
+#define LPCSTR char*
+#define DWORD long
+#define SIZE_T long
+
+/* link to import thunk */
+HMODULE __stdcall GetModuleHandleA (LPCSTR lpModuleName);
+/* link to directly to iat */
+__declspec(dllimport) HMODULE __stdcall LoadLibraryExA (LPCSTR lpLibFileName, HANDLE hFile, DWORD dwFlags);
+SIZE_T __stdcall VirtualQuery (LPCVOID lpAddress, LPCVOID lpBuffer, SIZE_T dwLength);
+
+#define PassWithoutSymbolOnWin PassWithoutSymbol
+#define PassWithDifferentAddressOnWin PassWithDifferentAddress
+#else
+#define PassWithoutSymbolOnWin Pass
+#define PassWithDifferentAddressOnWin Pass
+#endif
+
+#ifdef DLFCN_WIN32_SHARED
+#define PassWithSymbolOnSharedBuild Pass
+#else
+#define PassWithSymbolOnSharedBuild PassWithoutSymbol
+#endif
```

Comment:
So this is only to workround same issue as in https://github.com/dlfcn-win32/dlfcn-win32/pull/72#discussion_r507128585

Reply:
That issue will be solved (if possible) in another pull request